CMakeLists.txt

```diff
@@ -56,29 +59,45 @@ if(NOT GTest_FOUND)
 endif()
 
 add_library(FastBufferTree
-    buffer_tree.cpp
+    src/buffer_tree.cpp
     include/buffer_tree.h
-    buffer_control_block.cpp
+    src/buffer_control_block.cpp
     include/buffer_control_block.h
-    include/update.h sketch_write_manager.cpp include/sketch_write_manager.h include/sketch.h)
+    include/update.h 
+    src/sketch_write_manager.cpp 
+    include/sketch_write_manager.h 
+    include/sketch.h)
 target_link_libraries(FastBufferTree PRIVATE GTest::gtest)
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-  target_compile_options(FastBufferTree PRIVATE -O3)
-elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-  target_compile_options(FastBufferTree PRIVATE /O2)
+# optimize unless debug
+if (DEFINED ENV{DEBUG})
+  message("Disabling optimizations and enabling debug symbols")
+  target_compile_options(FastBufferTree PRIVATE -g)
+else ()
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    target_compile_options(FastBufferTree PRIVATE -O3)
+  elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+    target_compile_options(FastBufferTree PRIVATE /O2)
 endif()
+endif()
+# are we running in linux?
+if (UNIX AND NOT APPLE)
+  message("Enabling Fallocate for a linux system")
```

Comment:
what does fallocate do?

Reply:
It reserves a portion of disk space for a given file. fallocate is significantly faster than posix_fallocate though not portable.